qnapstats/qnap_stats.py

```diff
@@ -279,23 +279,47 @@ def get_bandwidth(self):
         """Obtain the current bandwidth usage speeds."""
         resp = self._get_url(
             "management/chartReq.cgi?chart_func=QSM40bandwidth",
-            force_list=("item")
+            force_list="item"
         )
 
         if resp is None:
             return None
 
-        details = {}
-
-        default = resp["bandwidth_info"]["df_gateway"]
+        if "bandwidth_info" not in resp:
+            # changes in API since QTS 4.5.4, old query returns no values
+            try:
+                resp = self._get_url("management/chartReq.cgi?chart_func=bandwidth")
+            except requests.exceptions.RequestException:
+                # no bandwidth_info and no new API means no way to return any results
+                return None
```

Comment:
Everything works, but this my my only doubt. With previous request handling, we only checked if the response is not `None`, and that was enough to proceed. Now, only with new request, we are explicitly catching a potential exception, but no longer checking if response is not `None`. **Maybe we could have a consistent handling of both requests**?

Also, my QNAP returns 200 regardless of what the `chart_func` query parameter value is, so in both cases there won't be any exception. Of course, there can be an exception if the other side doesn't know this API, any behaviour changes or QNAP is not available, but this will happen in case of both requests.

Reply:
Yes, you can add do 
`if resp and "bandwidth_info" not in resp`
And after move after your `_get_url` 
`if resp is None: return None`

Reply:
@julien-bouquet so skip catching exception right? nowhere else are network exceptions cought, so I think this would be consistent. It's also not needed as there shouldn't be any exception at this point. Is below ok?

```python
  resp = self._get_url(
      "management/chartReq.cgi?chart_func=QSM40bandwidth",
      force_list="item"
  )

  if resp and "bandwidth_info" not in resp:
      # changes in API since QTS 4.5.4, old query returns no values
      resp = self._get_url("management/chartReq.cgi?chart_func=bandwidth")

  if resp is None:
      return None
```

Reply:
You can keep catching exception to avoid `raise` but your sample is good

Reply:
I would prefer not to catch it becuase it's not done anywhere else in the code when doing API query, so this way we keep it consistent (maybe later someone can add catching exceptions to all calls, then it would be consistent too). One would think why is this place so special that the exception needs to be cought, while in fact, the exception should not be even thrown at this point becuase the API allows to fiddle with query params and providing unexpected values still ends with HTTP 200.